apps/avatax/src/modules/saleor/order.ts

```diff
@@ -47,3 +51,69 @@ export class SaleorOrder {
     return this.data.order.channel.taxConfiguration.pricesEnteredWithTax;
   }
 }
+
+interface ISaleorCancelledOrderEvent {
+  getChannelSlug(): string;
+  getAvataxId(): string;
+  getPrivateMetadata(): Array<{ key: string; value: string }>;
+}
+
+export class SaleorCancelledOrderEvent implements ISaleorCancelledOrderEvent {
+  private static schema = z.object({
+    order: z.object({
+      channel: z.object({
+        id: z.string(),
+        slug: z.string(),
+      }),
+      id: z.string(),
+      avataxId: z.string(),
+    }),
+    recipient: z.object({
+      privateMetadata: z.array(
+        z.object({
+          key: z.string(),
+          value: z.string(),
+        }),
+      ),
+    }),
+  });
+
+  private constructor(private data: z.infer<typeof SaleorCancelledOrderEvent.schema>) {}
+
+  getPrivateMetadata() {
+    return this.data.recipient.privateMetadata;
+  }
+
+  getChannelSlug() {
+    return this.data.order.channel.slug;
+  }
+
+  getAvataxId() {
+    return this.data.order.avataxId;
+  }
+
+  public static ParsingError = BaseError.subclass("AvataxAppSaleorOrderCancelledParsingError");
+
+  static create(payload: OrderCancelledPayloadFragment) {
+    if (!payload.order) {
+      return err(new OrderCancelPayloadOrderError("Insufficient order data"));
```

Comment:
you can pass "props" to second argument of modern-errors. We should put there Zod error. It will be visible when error is later sent to datadog/sentry. Now this error will tell us not too much when found